src/dnaio/_core.pyx

```diff
@@ -547,26 +513,25 @@ cdef class FastqIter:
             qualities_length = qualities_end - qualities_start
 
             # Check for \r\n line-endings and compensate
-            if self.buffer[name_end - 1] == b'\r':
+            if (name_end - 1)[0] == b'\r':
```

Comment:
I think switching to only pointers instead of indices is an improvement overall here, but I want to mention one downside as well: Previously, it was clear that `self.buffer[name_end - 1]` accesses an element of the buffer, and now that’s not the case anymore. One has to know from the context that `name_end` points into the buffer.

Being able to drop the `name_end_ptr`/`name_end` duality offsets this, though, IMO.

Reply:
Yes, I also think this is uglier than what was used previously. Luckily it still reads quite okay: one address back from the end of the name string.